web/src/features/AppVersionHistory/AppVersionHistoryRow.jsx

```diff
@@ -11,18 +11,19 @@ import { Utilities, getPreflightResultState } from "@src/utilities/utilities";
 import { YamlErrors } from "./YamlErrors";
 import Icon from "@src/components/Icon";
 
+import { ViewDiffButton } from "@features/VersionDiff/ViewDiffButton";
+
 class AppVersionHistoryRow extends Component {
-  renderDiff = (version) => {
-    const hideSourceDiff =
-      version.source?.includes("Airgap Install") ||
-      version.source?.includes("Online Install");
-    if (hideSourceDiff) {
-      return null;
-    }
-    return (
-      <div className="u-marginTop--5">{this.props.renderDiff(version)}</div>
-    );
-  };
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      // TODO: move this into selector
+      showViewDiffButton:
+        !props.version.source?.includes("Airgap Install") &&
+        !props.version.source?.includes("Online Install"),
+    };
+  }
```

Comment:
moved this logic into the state as `showViewDiffButton` to make it clearer what it's.